server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -59,6 +62,119 @@ public class IndexingStats implements Writeable, ToXContentFragment {
      */
     public static class Stats implements Writeable, ToXContentFragment {
 
+        /**
+         * Tracks item level rest category class codes during indexing
+         *
+         * @opensearch.internal
+         */
+        public static class DocStatusStats implements Writeable, ToXContentFragment {
+
+            final AtomicLong[] docStatusCounter;
+
+            public DocStatusStats() {
+                this(Stream.generate(AtomicLong::new).limit(5).toArray(AtomicLong[]::new));
+            }
+
+            protected DocStatusStats(final AtomicLong[] docStatusCounter) {
+                this.docStatusCounter = docStatusCounter;
+            }
+
+            public DocStatusStats(StreamInput in) throws IOException {
+                docStatusCounter = new AtomicLong[5];
+
+                for (int i = 0; i < docStatusCounter.length; ++i) {
+                    docStatusCounter[i] = new AtomicLong(in.readLong());
+                }
+            }
+
+            public void inc(final int familyCode) {
+                docStatusCounter[familyCode - 1].incrementAndGet();
+            }
+
+            public void add(final AtomicLong[] docStatusCounter) {
+                for (int i = 0; i < this.docStatusCounter.length; ++i) {
+                    this.docStatusCounter[i].addAndGet(docStatusCounter[i].longValue());
+                }
```

Comment:
Assert that the length of input array is within 5?
Also, array seems to be a leaky abstraction. While it may be fine to use array for internal implementation, do we need to expose this method or can we instead expose an (statusFamily, count) method?

Reply:
Yea, exposing it as a function makes more sense.